helm-charts/flink-historyserver/templates/historyserver.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: historyserver
+  name: historyserver
+spec:
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: historyserver
+  template:
+    metadata:
+      labels:
+        app: historyserver
+    spec:
+      containers:
+      - args:
+        - history-server
+        command:
+        - /docker-entrypoint.sh
+        env:
+        - name: _POD_IP_ADDRESS
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: status.podIP
+        image: flink:{{- .Values.flinkVersion }}
+        name: flink-main-container
+        ports:
+        - containerPort: 8082
+          name: history
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 300m
+            memory: 1536Mi
+          requests:
+            cpu: 300m
+            memory: 1536Mi
+        startupProbe:
+          httpGet:
+            path: /config
+            port: history
+        volumeMounts:
+        - mountPath: /opt/history/jobs
+          name: efs-flink-history
+        - mountPath: /opt/flink/conf
+          name: flink-config-volume
+      initContainers:
+      - command:
+        - sh
+        - -c
+        - chown 9999:9999 /opt/history/jobs && ls -lhd /opt/history/jobs
+        image: busybox:1.36.1
+        imagePullPolicy: IfNotPresent
+        name: efs-mount-ownership-fix
+        resources: {}
+        volumeMounts:
+        - mountPath: /opt/history/jobs
+          name: efs-flink-history
+      schedulerName: default-scheduler
+      securityContext:
+        fsGroup: 9999
+      # NOTE: this SA is set up by the flink-operator helm chart and reused here
+      serviceAccountName: flink
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - name: efs-flink-history
+        persistentVolumeClaim:
+          claimName: flink-historyserver-efs-pvc
+      - configMap:
+          items:
+          - key: log4j-console.properties
+            path: log4j-console.properties
+          - key: flink-conf.yaml
+            path: flink-conf.yaml
+          # NOTE: this configmap is set up by the flink-operator helm chart and reused here
+          name: flink-operator-config
+        name: flink-config-volume
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: historyserver
+  name: historyserver-rest
+spec:
+  ports:
+  - name: history
+    port: 8082
+    targetPort: 8082
+  selector:
+    app: historyserver
+  type: ClusterIP
+status:
```

Comment:
`status` fields are written out by the apiserver, we shouldn't have them in helm charts